src/magic_folder/config.py

```diff
@@ -1123,6 +1124,32 @@ def get_currentsnapshot_pathstate(self, cursor, name):
                 return PathState(mtime_ns=row[0], ctime_ns=row[1], size=row[2])
             raise KeyError(name)
 
+    @with_cursor
+    def get_all_current_snapshot_pathstates(self, cursor):
+        """
+        Return the PathState for every file we have in our
+        'current_snapshot' table.
+
+        :returns Iterable[(unicode, PathState)]: an iterable of
+            2-tuples of (name, PathState instance), one for each file
+        """
+        cursor.execute(
+            """
+            SELECT
+                name, mtime_ns, ctime_ns, size
+            FROM
+                current_snapshots
+            ORDER BY
+                mtime_ns DESC
```

Comment:
It is probably worth a mention in the docstring about this ordering. My guess of ordering would be by name, rather than mtime.